xconnect-searchstax-connector-9.1.1.ps1

```diff
@@ -0,0 +1,169 @@
+         Write-Error -Message "Unable to get Auth Token. Error was: $_" -ErrorAction Stop
+    }
+}
+
+function Check-DeploymentExist($token) {
+    try {
+        $headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+        $headers.Add("Authorization", "Token $token")
+        $result = Invoke-WebRequest -Method Get -Headers $headers -uri $deploymentReadUrl
+        if ($result.statuscode -eq 200) {
+            Write-Host "Deployment found. Continuing."
+        } else {
+            Write-Error "Could not find deployment. Exiting." -ErrorAction Stop
+        }
+    } catch {
+        Write-Error -Message "Unable to verify if deployment exists. Error was: $_" -ErrorAction Stop
+    }
+}
+
+function Upload-Config($solrVersion, $token) {
+    try {
+        $headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+        $headers.Add("Authorization", "Token $token")
+
+        if ($solrVersion -eq "7.2.1") {
+            $form = @{
+                name = "sitecore-xdb_$sitecorePrefix"
+                files = Get-Item -Path $solrConfigPath721
+            }
+            Invoke-RestMethod -Method Post -Form $form -Headers $headers -uri $configUploadUrl 
+        }
+    } catch {
+        Write-Error -Message "Unable to upload config file. Error was: $_" -ErrorAction Stop
+    }    
+}
+
+function Get-Node-Count($token) {
+    $headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+    $headers.Add("Authorization", "Token $token")
+    $result = Invoke-RestMethod -Method Get -ContentType 'application/json' -Headers $headers -uri $deploymentReadUrl
+    return [int]$result.num_nodes_default + [int]$result.num_additional_app_nodes
+}
+
+function Get-SolrUrl($token) {
+    $headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+    $headers.Add("Authorization", "Token $token")
+    $result = Invoke-RestMethod -Method Get -ContentType 'application/json' -Headers $headers -uri $deploymentReadUrl
+    return $result.http_endpoint
+}
+
+#TODO : Too many moving parts - Add try-catch blocks and make it fault tolerant
+function Create-Collections($solrVersion, $token) {
+    "Getting live node count ..."
+    $nodeCount = Get-Node-Count $token
+    "Getting live node count ... DONE"
+    "Number of nodes - $nodeCount"
+    $solr = Get-SolrUrl $token
+    Write-Host $solr
+    if ($solrUsername.length -gt 0){
+        $secpasswd = ConvertTo-SecureString $solrPassword -AsPlainText -Force
+        $credential = New-Object System.Management.Automation.PSCredential($solrUsername, $secpasswd)
+    }
+    "Creating Collections ... "
+    foreach($collection in $collections){
+        $collection | Write-Host
+        if ($solrVersion -eq "7.2.1") {
+            $url = -join($solr, "admin/collections?action=CREATE&name=",$sitecorePrefix,$collection,"&numShards=1&replicationFactor=",$nodeCount,"&collection.configName=sitecore-xdb1_$sitecorePrefix")
```

Comment:
this needs to be renamed from "sitecore-xdb1_" to "sitecore-xdb" - Which should match with original uploaded config file

Reply:
Rest all is good

Reply:
This is now done.